packages/shared/hooks/useEscape.ts

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useCallback, useEffect } from 'react';
+
+function useEscape(handler) {
+  const handlerMemo = useCallback(handler, [handler]);
```

Comment:
i'm curious why we `usecallback` here? wouldn't `handler` reference change every time the caller re-renders? 

in LabelSelector it's used like this:
```

  useClickOutside(addLabelContainerRef, () => {
    setShowAdd(false);
  });

  useEscape(() => {
    setShowAdd(false);
  });
```

Reply:
Good catch, I removed that handling from here and put the `useCallback` in `LabelSelector`.